File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerConfig.java
package com.brentvatne.exoplayer;

import androidx.media3.exoplayer.upstream.DefaultBandwidthMeter;
import androidx.media3.exoplayer.upstream.LoadErrorHandlingPolicy;

/**
 * Extension points to configure the Exoplayer instance
 */
public interface ReactExoplayerConfig {
    LoadErrorHandlingPolicy buildLoadErrorHandlingPolicy(int minLoadRetryCount);

    void setDisableDisconnectError(boolean disableDisconnectError);
    boolean getDisableDisconnectError();

    DefaultBandwidthMeter getBandwidthMeter();
}

File: android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerConfig.kt
package com.brentvatne.exoplayer

import androidx.media3.exoplayer.upstream.DefaultBandwidthMeter
import androidx.media3.exoplayer.upstream.LoadErrorHandlingPolicy

interface ReactExoplayerConfig {
    fun buildLoadErrorHandlingPolicy(minLoadRetryCount: Int): LoadErrorHandlingPolicy
    var disableDisconnectError: Boolean
    val bandwidthMeter: DefaultBandwidthMeter
    var initialBitrate: Long?
    fun setInitialBitrate(bitrate: Long)
}
